### PR TITLE
Fuse Hector SLAM, Wheel odometry, and ORB_SLAM3 poses

### PR DESCRIPTION
**Pull request by hello-atharva**
## Description

This PR implements a basic averaging Kalman filter to fuse multiple pose estimates.

TODOs:
1. Fix a regression for in-place rotations observed in ORB_SLAM3.
2. Implement full system dynamic model.
3. Add redundancy in case ORB_SLAM fails to localize.

## Checklist

- \[ \] I have performed a self-review of my code
- \[ \] If it is a core feature, I have added thorough tests
- \[ \] I have added documentation for the changes
- \[ \] I have updated the README file if necessary
- \[ \] I have run on hardware if necessary

## Screenshots (if applicable)

Add any relevant screenshots or screen recordings to help reviewers understand the changes.

## Additional context

Add any other context or information about the pull request here.

### CODE REVIEW

File: src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/state_estimator.py [hunk 1]
```diff
@@ -66,6 +68,61 @@ def __init__(self, trust_slam: bool = False, use_history: bool = True):
...
+        self.H = np.eye(6)
+
+        dt = 0.1 # TODO: get from odom message
+        self.F = np.eye(8)
+
+        self.measurement1 = None
+        self.measurement2 = None
+        self.measurement3 = None
+
+    def predict_kalman(self):
```
Comment (cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/state_estimator.py:85]: Please add a bit more documentation to these

File: src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/state_estimator.py [hunk 2]
```diff
@@ -66,6 +68,61 @@ def __init__(self, trust_slam: bool = False, use_history: bool = True):
...
+        self.x = self.F @ self.x
+        self.P = self.F @ self.P @ self.F.T + self.Q
+
+    def update_kalman(self, z):
```
Comment (cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/state_estimator.py:89]: more docts + type hints ; what is z especially? what does this do?

File: src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/state_estimator.py [hunk 3]
```diff
@@ -66,6 +68,61 @@ def __init__(self, trust_slam: bool = False, use_history: bool = True):
...
+        y = z - (self.H @ self.x)
+        S = self.H @ self.P @ self.H.T + self.R
+        K = self.P @ self.H.T @ np.linalg.inv(S)
+        
+        self.x = self.x + K @ y
+        self.P = (np.eye(6) - K @ self.H) @ self.P
+
+    def fuse_measurements(self, wheel_measurement, slam_measurement, vio_measurement = None):
```
Comment (cpaxton) [inline src/stretch_ros2_bridge/stretch_ros2_bridge/nodes/state_estimator.py:97]: docs etc

**Review by cpaxton — COMMENTED** [review]
We should:
- add some more documentation to the code
- add an install guide under docs/ in order to help people get started with Orbslam

**Comment by cpaxton** [comment]
Also please run some pre-commit hooks on this code

**Comment by hello-cpaxton** [comment]
LGTM - some possible issues when running mapping that need to be resolved though

**Review by hello-cpaxton — APPROVED** [review]
lgtm, todo make mapping app work

**Pull request merged**